enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+class Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      if is_a? Hash
+        yield(keys[i], self[keys[i]])
+        return self if i == size - 1
+      elsif is_a? Array
+        yield self[i]
+        return self if i == size - 1
+      elsif is_a? Range
+        yield to_a[i]
+        return self if i == size - 1
+      end
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      if is_a? Hash
+        yield(keys[i], self[keys[i]])
+        return self if i == size - 1
+      elsif is_a? Array
+        yield(self[i], i)
+        return self if i == size - 1
+      elsif is_a? Range
+        yield to_a[i], i
+        return self if i == size - 1
+      end
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
```

Comment:
* [ ] should have working my_select identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
```p (1..3).select{|a| a.even? }```  should be equal to ```p (1..3).my_select{|a| a.even? }```